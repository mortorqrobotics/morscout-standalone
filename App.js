/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from "react";
// import Message, { RandomMessage } from '~/Basic/Messages';
import { Platform, StyleSheet, Text, View } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#F5FCFF",
  },
  welcome: {
    fontSize: 20,
    textAlign: "center",
    margin: 10,
  },
  instructions: {
    textAlign: "center",
    color: "#333333",
    marginBottom: 5,
  },
});

const instructions = Platform.select({
  ios: `Press Cmd+R to reload,
    Cmd+D or shake for dev menu`,
  android:
    "Double tap R on your keyboard to reload,\n" +
    "Shake or press menu button for dev menu",
});

export default () => (
  <View style={styles.container}>
    {/* <Message style={styles.welcome}>Welcome to React Native!</Message> */}
    <Text style={styles.instructions}>To get started, edit App.js</Text>
    <Text style={styles.instructions}>{instructions}</Text>
  </View>
);
