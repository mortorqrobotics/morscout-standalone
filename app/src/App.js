import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import Matches from 'screen/Matches';
import Match from 'screen/Match';
import Index from 'screen/Index';
import Settings from 'screen/Settings';
import { Route, Router, Switch } from '@/react-router';
import { View, Text } from 'react-native';
import H from '@/H';
import Navigation from '~/TopNav';
import store from 'store';
import initialize from 'actions/Basic/init';

const unsub = store.subscribe(() => store.getState());
store.dispatch(initialize);
unsub();

// eslint-disable-next-line
function dataDumper(props){
  return (
    <H level={6}>
      {JSON.stringify(props)}
    </H>
  );
}


// eslint-disable-next-line
class BackButton extends React.Component {
  static contextTypes = {
    // eslint-disable-next-line
    router: PropTypes.object,
  };

  render() {
    const {
      context,
    } = this;
    return (
      <Text onClick={context.router.history.goBack}>
        Back
      </Text>
    );
  }
}

function NotFoundError() {
  return (
    <View>
      <H level={1}>
        Page not found
      </H>
      <BackButton />
    </View>
  );
}

export default () => (
  <Provider store={store}>
    <Router>
      <View>
        <Navigation />
        <Switch>
          <Route path="/index" exact component={Index} />
          <Route path="/" exact component={Index} />
          <Route path="/Matches" exact component={Matches} />
          <Route path="/Profile" exact component={Settings} />
          <Route path="/Match/:id" component={Match} />
          <Route component={NotFoundError} />
        </Switch>
      </View>
    </Router>
  </Provider>
);
