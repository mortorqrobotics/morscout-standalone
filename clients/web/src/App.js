import React from "react";
import { Provider } from "react-redux";
import Index from "screen/index";
import Matches from "screen/matches";
import Match from "screen/match";
import Teams from "screen/teams";
import { createNavigator, SwitchRouter } from "@react-navigation/core";
import { createBrowserApp } from "@react-navigation/web";
import Navigation from "@/navigation";
import store from "client-common-redux";

const App = createBrowserApp(
  createNavigator(
    Navigation,
    SwitchRouter({
      Index,
      Matches,
      Match,
      Teams
    }),
    {}
  )
);

export default () => (
  <Provider store={store}>
    <App />
  </Provider>
);
