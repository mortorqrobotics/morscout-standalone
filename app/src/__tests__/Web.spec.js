// __tests__/Loader.spec.js
import React from "react";
import renderer from "react-test-renderer";
import App from "../App.web";

jest.mock("~/Basic/Messages");

test("Match renders Correctly", async () => {
  expect.assertions(1);
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
