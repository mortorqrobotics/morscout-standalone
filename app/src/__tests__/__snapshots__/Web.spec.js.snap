// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match renders Correctly 1`] = `
Array [
  <div
    className="ur ur-9a09276 rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <div
      className="ur ur-d4865e9 rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <style>
          .ur.ur-487a89d{background-color:orange;flex-direction:column;}
        </style>
        MorScout
      </a>
      <a
        href="/Matches"
        onClick={[Function]}
      >
        <style>
          .ur.ur-487a89d{background-color:orange;flex-direction:column;}
        </style>
        All Matches
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <style>
          .ur.ur-487a89d{background-color:orange;flex-direction:column;}
        </style>
        Team List
      </a>
      <style>
        .ur.ur-d4865e9{flex-basis:auto;flex-direction:row;flex-grow:1;flex-shrink:1;}
      </style>
    </div>
    <div
      className="ur ur-89cb11e rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    >
      <div
        className="ur ur-487a89d rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
        css={
          Object {
            "backgroundColor": "orange",
            "flexDirection": "column",
          }
        }
        dir="auto"
        onClick={[Function]}
      >
        <style>
          .ur.ur-487a89d{background-color:orange;flex-direction:column;}
        </style>
        MorTeam
      </div>
      <a
        href="/"
        onClick={[Function]}
      >
        <style>
          .ur.ur-487a89d{background-color:orange;flex-direction:column;}
        </style>
        Profile
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <style>
          .ur.ur-487a89d{background-color:orange;flex-direction:column;}
        </style>
        Log Out
      </a>
      <style>
        .ur.ur-89cb11e{flex-basis:auto;flex-direction:row;flex-grow:1;flex-shrink:1;justify-content:flex-end;}
      </style>
    </div>
    <style>
      .ur.ur-9a09276{align-items:flex-start;flex-direction:row;flex-wrap:wrap;justify-content:space-between;}
    </style>
  </div>,
  <div
    className="ur ur-99914b9 rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <h1>
      MorScout -- Index
    </h1>
    <div
      className="Message rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
      onClick={[Function]}
    >
      <span
        className="Message rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-poiln3 rn-fontSize-7cikom rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-irrty rn-wordWrap-qvutc0"
        dir="auto"
      >
        Made With Fifty Shades of Orange
      </span>
    </div>
    <style>
      .ur.ur-99914b9{}
    </style>
  </div>,
]
`;
